src/frontend/src/lib/services/loader.services.ts

```diff
@@ -0,0 +1,37 @@
+import { allowSigning } from '$lib/api/backend.api';
+import { errorSignOut } from '$lib/services/auth.services';
+import { authStore } from '$lib/stores/auth.store';
+import { i18n } from '$lib/stores/i18n.store';
+import type { ResultSuccess } from '$lib/types/utils';
+import { get } from 'svelte/store';
+
+/**
+ * Initializes the signer allowance by calling `allow_signing`.
+ *
+ * This function should be called once during boot time before retrieving ETH or BTC addresses.
+ * It allocates a cycles budget sufficient for a reasonable number of signer calls.
+ * A "reasonable" number is currently defined as 30 calls, allowing the user to retrieve their ETH and BTC addresses
+ * and perform up to 28 additional transactions.
+ *
+ * If an error occurs during the `allow_signing` call, the user will be signed out,
+ * as the Oisy Wallet cannot function without ETH or Bitcoin addresses.
+ *
+ * @async
+ * @function initSignerAllowance
+ * @returns {Promise<ResultSuccess>} Returns an object indicating success or failure of the operation.
+ * @throws Will trigger a sign-out if `allow_signing` fails.
+ */
+export const initSignerAllowance = async (): Promise<ResultSuccess> => {
+	try {
+		const { identity } = get(authStore);
+
+		await allowSigning({ identity });
+	} catch (err: unknown) {
+		// In the event of any error, we sign the user out, as we assume that the Oisy Wallet cannot function without ETH or Bitcoin addresses.
+		await errorSignOut(get(i18n).init.error.allow_signing);
```

Comment:
would it make sense to call the `signOut` function here? to eventually clean the storage, etc (in the same way we do in case the loading of address is not successful)

Reply:
not sure what you mean? it's literally what `errorSignOut` does.